Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/Model/Building/SqlStorageTypeInformationProvider.cs

```diff
@@ -226,6 +229,42 @@ public IStorageTypeInformation GetStorageType (object? value)
       return null;
     }
 
+    private StorageTypeInformation GetDateTimeStorageType (bool isNullableInDatabase, Type dotNetType, IPropertyInformation? propertyInformation)
+    {
+      if (propertyInformation != null)
+      {
+        var storageTypeFromPropertyInfo = GetDateTimeStorageTypeFromPropertyInformation(propertyInformation, isNullableInDatabase, dotNetType);
+
+        if (storageTypeFromPropertyInfo != null)
+          return storageTypeFromPropertyInfo;
+      }
+
+      return new StorageTypeInformation(typeof(DateTime), "datetime2", DbType.DateTime2, isNullableInDatabase, null, dotNetType, new DefaultConverter(dotNetType));
+    }
+    private StorageTypeInformation? GetDateTimeStorageTypeFromPropertyInformation (IPropertyInformation propertyInfo, bool isNullableInDatabase, Type dotNetType)
+    {
+      var dateTimeStorageTypeAttribute = propertyInfo.GetCustomAttribute<DateTimeStorageTypeAttribute>(inherited: false);
+
+      if (dateTimeStorageTypeAttribute == null)
+        return null;
+
+      var (dbType, storageTypeName) = dateTimeStorageTypeAttribute.StorageType switch
+      {
+        DateTimeStorageType.DateTime => (DbType.DateTime, "datetime"),
+        DateTimeStorageType.DateTime2 => (DbType.DateTime2, "datetime2"),
+        _ => throw ArgumentUtility.CreateEnumArgumentOutOfRangeException(nameof(dateTimeStorageTypeAttribute.StorageType), dateTimeStorageTypeAttribute.StorageType)
```

Comment:
I think this is typically done with an InvalidOperationException ("Enum value '{0}' is not valid for nameof(DateTimeStorageType")